docs/appendices/release-notes/4.2.0.rst

```diff
@@ -150,13 +150,13 @@ SQL Standard and PostgreSQL compatibility improvements
   :ref:`server_version <conf-session-server_version>` read-only session
   settings.
 
-- Added the `pg_catalog.pg_proc <postgres_pg_catalog>`_ table.
+- Added the :ref:`pg_catalog.pg_proc <postgres_pg_catalog>` table.
```

Comment:
Why is it now necessary to use `:ref:` in some places, but not in all? 

https://github.com/crate/crate/pull/10831/files#diff-ed2a6369cdf047f097bfca45fa6b4367d8a969567bdbfdfc3fd744d55af1a516R43 seems to still be working without it?

Reply:
Inline links like \`text <data-type-text_>\`_ point to shortcuts like `.. _data-type-text` within the same file (usually), but `:ref:` links resolve against the whole reference database across the whole tree.

Many of them have been wrong and did not resolve properly but have apparently not been detected by the linkchecker of Sphinx 1.x. For example, try these at [1]:
- Added the `pg_catalog.pg_proc` table.
- Added the `pg_catalog.pg_range` table.
- Added the `pg_catalog.pg_enum` table.

[1] https://crate.io/docs/crate/reference/en/4.3/appendices/release-notes/4.2.0.html#sql-standard-and-postgresql-compatibility-improvements

Reply:
this is an important point, @amotl 

links to internal labels should always be styled:

```
:ref:"link text <label-text>"
```

this makes it clear they are internal links, and it also activates a different sort of internal cross-reference checking that does not take place if you are linking to bare URLs (absolute or relative)

links to external URLs should always be styled:

```
`link text`_

[...]

.. _link text: http://example.com/
```

never inline, like this:

```
`link text <http://example.com/>`
```

if you wanted to link to http://example.com/, but didn't want to duplicate the URL at the bottom of the file. you can do this:

```
.. _link text: http://example.com/
.. _different link text: `link text`_
```

(yes you read that right, the second line defines the link by pointing to the previous link. @seut taught me this trick and it is awesome)

then you do:

```
`link text`_, `different link text`_
```

so the style for external links remains consistent

the stuff that I wrote, and the stuff that you wrote, combined would make a useful addition to https://github.com/crate/crate-docs/blob/master/style/rst.rst, I think. would you like to draft a PR, @amotl, or should I? happy to collab <3

Reply:
> the stuff that I wrote, and the stuff that you wrote, combined would make a useful addition to https://github.com/crate/crate-docs/blob/master/style/rst.rst, I think. would you like to draft a PR, @amotl, or should I? happy to collab <3

I don't think we need to reinvent the rst doc or sphinx doc. We just need to get the link checker working correctly.